Mathlib/CategoryTheory/LocallyCartesianClosed/Prelim.lean

```diff
@@ -0,0 +1,379 @@
+    apply pullback.hom_ext <;> simp
+    · exact congr_arg CommaMorphism.left (h ⟨ .right⟩)
+    · exact congr_arg CommaMorphism.left (h ⟨ .left ⟩)
+
+attribute [local instance] CartesianMonoidalCategory.ofFiniteProducts
+
+instance (X : C) : CartesianMonoidalCategory (Over X) := by
+  infer_instance
+
+/-- The object `sigma Y (reindex Y Z)` is isomorphic to the binary product `Y ⊗ Z`
+in `Over X`. -/
+@[simps!]
+def sigmaReindexIsoProd (Y Z : Over X) :
+    sigma Y (reindex Y Z) ≅ Y ⊗ Z := by
+  apply IsLimit.conePointUniqueUpToIso (isBinaryProductSigmaReindex Y Z) (prodIsProd Y Z)
+
+/-- Given a morphism `f : X' ⟶ X` and an object `Y` over `X`, the object
+`(map f).obj ((pullback f).obj Y)` is isomorphic to the binary product of `Over.mk f` and `Y`. -/
+def sigmaReindexIsoProdMk {Y : C} (f : Y ⟶ X) (Z : Over X) :
+    (map f).obj ((pullback f).obj Z) ≅ Over.mk f ⊗ Z :=
+  sigmaReindexIsoProd (Over.mk f) _
+
+@[reassoc (attr := simp)]
+lemma sigmaReindexIsoProd_hom_comp_fst {Y Z : Over X} :
+    (sigmaReindexIsoProd Y Z).hom ≫ fst Y Z = π_ Y Z :=
+  IsLimit.conePointUniqueUpToIso_hom_comp
+    (isBinaryProductSigmaReindex Y Z) (Limits.prodIsProd Y Z) ⟨.left⟩
+
+@[reassoc (attr := simp)]
+lemma sigmaReindexIsoProd_hom_comp_snd {Y Z : Over X} :
+    (sigmaReindexIsoProd Y Z).hom ≫ snd Y Z = μ_ Y Z :=
+  IsLimit.conePointUniqueUpToIso_hom_comp
+    (isBinaryProductSigmaReindex Y Z) (Limits.prodIsProd Y Z) ⟨.right⟩
+
+end BinaryProduct
+
+end Over
+
+section TensorLeft
+
+open MonoidalCategory Over Functor CartesianMonoidalCategory
+
+attribute [local instance] CartesianMonoidalCategory.ofFiniteProducts
+
+variable [HasFiniteWidePullbacks C] {X : C}
+
+/-- The pull-push composition `pullback Y.hom ⋙ map Y.hom` is naturally isomorphic
+to the left tensor product functor `Y ⊗ _` in `Over X`. -/
+def Over.sigmaReindexNatIsoTensorLeft (Y : Over X) : pullback Y.hom ⋙ map Y.hom ≅ tensorLeft Y :=
+  NatIso.ofComponents
+    (fun Z => sigmaReindexIsoProd Y Z)
+    (by
+      intro Z Z' f
+      simp
+      ext1 <;> simp_rw [assoc]
+      · simp_rw [whiskerLeft_fst]
+        iterate rw [sigmaReindexIsoProd_hom_comp_fst]
+        ext
+        simp
+      · simp_rw [whiskerLeft_snd]
+        iterate rw [sigmaReindexIsoProd_hom_comp_snd, ← assoc, sigmaReindexIsoProd_hom_comp_snd]
+        ext
+        simp [reindex.sndProj])
+
+@[simp]
+lemma Over.sigmaReindexNatIsoTensorLeft_hom_app {Y : Over X} (Z : Over X) :
+    (Over.sigmaReindexNatIsoTensorLeft Y).hom.app Z = (sigmaReindexIsoProd Y Z).hom := by
+  aesop
+
+end TensorLeft
+
+/-- The functor from `C` to `Over (⊤_ C)` which sends `X : C` to `terminal.from X`. -/
+@[simps! obj_left obj_hom map_left]
+def Functor.toOverTerminal (X : C) (h : IsTerminal X) : C ⥤ Over X where
+  obj X := Over.mk <| h.from X
+  map {X Y} f := Over.homMk f
+
+/-- The slice category over the terminal object is equivalent to the original category. -/
+def equivOverTerminal (X : C) (h : IsTerminal X) : Over (X) ≌ C where
```

Comment:
Could you move this to the `IsTerminal` namespace? Also, the `inverse` functor could probably be "inlined" in the definition of the equivalence. Could you also add the `simps` attribute?